src/com/ConnectionInfoPublisher.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ConnectionInfoPublisher.hpp"
+#include <chrono>
+#include <thread>
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/filesystem.hpp>
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 int rank,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  rank(rank),
+  addressDirectory(addressDirectory)
+{}
+  
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  addressDirectory(addressDirectory)
+{}
+
+
+std::string ConnectionInfoPublisher::getFilename()
+{
+  using namespace boost::filesystem;
+  
+  constexpr int firstLevelLen = 2;
+  boost::uuids::string_generator ns_gen;
+  auto ns = ns_gen("af7ce8f2-a9ee-46cb-38ee-71c318aa3580"); // md5 hash of precice.org as namespace
+  
+  boost::uuids::name_generator gen{ns};
+  std::string s = acceptorName + requesterName + std::to_string(rank);
```

Comment:
```suggestion
const  std::string s = acceptorName + requesterName + std::to_string(rank);
```

Reply:
Sorry, these type of `const`s in a function of less 15 lines or so make zero sense for me. There is no performance gained, it only adds to human parsing complexity when reading the code.

Reply:
True in this case.

Yes, `const` adds to the complexity of humans parsing the code, but it reduces the complexity of understanding the code and prevents others from breaking your code in the future.
That's why I am trying to push it as much as possible.